shell/server/proxy.js

```diff
@@ -416,8 +420,27 @@ var getProxyForHostId = function (hostId) {
       return inMeteor(function () {
         var session = Sessions.findOne({hostId: hostId});
         if (!session) {
-          // Does not appear to be a valid session host.
-          return undefined;
+          if (isAlreadyOpened) {
+            return new Promise(function (resolve, reject) {
+              inMeteorInternal(function () {
```

Comment:
Hmm, inMeteorInternal isn't meant to be called directly.

Is it actually needed? I thought that the function passed into Promise's constructor was called synchronously, so would already be in the parent's context.